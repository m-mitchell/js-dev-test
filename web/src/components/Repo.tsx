import React, { useState } from 'react';
import moment from 'moment'

interface RepoProps {
  repoData: any
}
interface Commit {
  sha?: string
  commit?: any
}
const Repo: React.FC<RepoProps> = ({ repoData }) => {
  const [commit, setCommit]: [Commit, any] = useState({});
  const getCommit = async () => {
    if (commit.sha) {
      return;
    }
    try {
      const res = await fetch(`https://api.github.com/repos/${repoData.full_name}/commits`);
      const json = await res.json();
      if (json.length) {
        setCommit(json[0]);
      }
    } catch (err) {
      console.log('error', err);
    }
  }

  return (
    <div className="repo" onClick={() => getCommit()}>
      <h3>{repoData.name}</h3>
      <p>{repoData.description}</p>
      <dl>
        <dt>Language</dt>
        <dd>{repoData.language}</dd>
        <dt>Fork count</dt>
        <dd>{repoData.forks}</dd>
      </dl>
      { commit.sha
        && (
          <>
            <h4>Latest Commit</h4>
            <p>
              {commit.commit!.message}
              <br />
              <span className="commit-info">
                {commit.commit!.author.name} committed at {moment(commit.commit!.author.date).format('MM/DD/YY h:mm:ss a')}
              </span>
            </p>
          </>
        )
      }
    </div>
  );
};

export default Repo;
