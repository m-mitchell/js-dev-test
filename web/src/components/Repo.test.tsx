/* eslint-env jest */

import React from 'react';
import renderer from 'react-test-renderer';
import Repo from './Repo';

const exampleRepo = {
  'id': 140745386,
  'name': 'accessible-google-places-autocomplete',
  'full_name': 'silverorange/accessible-google-places-autocomplete',
  'description': null,
  'fork': false,
  'language': 'TypeScript',
  'forks': 3,
};

test('renders correctly', () => {
  const tree = renderer
    .create(<Repo repoData={exampleRepo} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
