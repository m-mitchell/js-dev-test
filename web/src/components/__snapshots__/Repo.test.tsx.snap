// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="repo"
  onClick={[Function]}
>
  <h3>
    accessible-google-places-autocomplete
  </h3>
  <p />
  <dl>
    <dt>
      Language
    </dt>
    <dd>
      TypeScript
    </dd>
    <dt>
      Fork count
    </dt>
    <dd>
      3
    </dd>
  </dl>
</div>
`;
